scene/3d/collision_shape_3d.cpp

```diff
@@ -68,46 +68,69 @@ void CollisionShape3D::make_convex_from_siblings() {
 	set_shape(shape_new);
 }
 
+CollisionObject3D *CollisionShape3D::_get_ancestor_collision_object() const {
+	Node *parent = get_parent();
+	while (parent) {
+		CollisionObject3D *co = Object::cast_to<CollisionObject3D>(parent);
+		if (co) {
+			return co;
+		}
+		parent = parent->get_parent();
+	}
+	return nullptr;
+}
+
+Transform3D CollisionShape3D::_get_transform_to_collision_object() const {
+	Transform3D transform_to_col_obj = get_transform();
+	Node *parent = get_parent();
+	while (parent != collision_object) {
+		Node3D *parent_3d = Object::cast_to<Node3D>(parent);
```

Comment:
This code is making a lot of assumptions. This isn't only called on creation: it is also called later, such as `NOTIFICATION_LOCAL_TRANSFORM_CHANGED`. The parent object could be destroyed or nullptr, such as if some node got free'd since last frame, or there might no longer be a collision_object ancestor, and the cast_to should be checked in case a non-Node3D exists.

Reply:
Thanks, added a check. I also added `unlikely()` around it since we assume that this is an edge case.